### PR TITLE
fix: Tree filter node style lost

### LINKED ISSUE #49767: Tree filterTreeNode The filtered nodes are not highlighted
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-frjbpt-sk3dix?file=demo.tsx)

image

### Steps to reproduce

搜索任意节点

### What is expected?

高亮

### What is actually happening?

没有高亮

| Environment | Info |
| --- | --- |
| antd | 5.19.1 |
| React | 18.3.1 |
| System | Windows |
| Browser | Chrome 123.0.6312.86 |

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- fix https://github.com/ant-design/ant-design/issues/49767

### 💡 Background and solution

tree 组件 filter-node 缺少高亮

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix: Tree filter node style lost      |
| 🇨🇳 Chinese |      修复 tree 组件 filter-node 节点高亮样式丢失     |

### CODE REVIEW

**Comment by wanpan11** [comment]
以前的好像是有高亮

image

**Comment by afc163** [comment]
图片

会不会感觉有点重？

File: components/tree/style/index.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ export const genBaseStyle = (prefixCls: string, token: TreeToken): CSSObject =>
         },
 
         [`&:not(${treeNodeCls}-disabled).filter-node ${treeCls}-title`]: {
-          color: 'inherit',
+          color: token.colorPrimary,
```
Comment (afc163) [inline components/tree/style/index.ts:190]: 之前是在 https://github.com/ant-design/ant-design/pull/28984 是这里改掉的。

**Comment by wanpan11** [comment]
> 会不会感觉有点重？

TreeSelect 组件下搜索有过滤，不用高亮应该也行。
Tree 组件下的搜索感觉可以加，不然看不到效果 😂

image

**Comment by wanpan11** [comment]
看官网的这个 [demo](https://ant.design/components/tree-cn#components-tree-demo-search) , 是不是推荐这样做搜索功能，才改成继承用户传进来的颜色

**Comment by wanpan11** [comment]
看官网是推荐用这个 [demo](https://ant.design/components/tree-cn#components-tree-demo-search) 来实现搜索的，有过滤节点的话感觉高不高亮都行

**Review by afc163 — APPROVED** [review]

**Pull request merged**